### ISSUE TITLE
Push stdlib version to the stdlib-fpm branch

### ISSUE THREAD

**Issue by perazz** [fortran-lang/stdlib#701]
### Motivation

Hello stdlib developers, 

apologies I'm no stdlib expert yet.

I think it would be fantastic to soon see tighter/full integration of stdlib in fpm, meaning, that fpm could enable stdlib as a meta-package with basically no user input (like any C/C++ compiler does for their stdlibs). 

I see that the CI does some [minimal cleanup of the sources](https://github.com/fortran-lang/stdlib/blob/master/ci/fpm-deployment.sh) before pushing to the stdlib-fpm branch. I would like to: 

- ask whether there is any interest in it? 

- ask for feedback about whether any of those pre-processing steps could instead be replaced with more advanced source handling in fpm;

- fpm.toml does not store a `VERSION`, is that by design? otherwise, I'd like to suggest a PR that pushes it on deployment: 

https://github.com/fortran-lang/stdlib/blob/261a6274477ec8e23f94bd8be384158149ed6628/fpm.toml#L1-L10

**Comment by jvdp1** [comment]
I totally agree with you.
During the GSoC 2022, @arteevraina, together with @awvwgk, worked on supporting preprocessing with `fpm` which would enable `fpm` to compile `stdlib`. See this [PR](https://github.com/fortran-lang/fpm/pull/729) for more details. Once preprocessing is supported by `fpm`, it should be able to fully compile `stdlib`.

**Comment by arteevraina** [comment]
> ### Motivation
> Hello stdlib developers,
> 
> apologies I'm no stdlib expert yet.
> 
> I think it would be fantastic to soon see tighter/full integration of stdlib in fpm, meaning, that fpm could enable stdlib as a meta-package with basically no user input (like any C/C++ compiler does for their stdlibs).
> 
> I see that the CI does some [minimal cleanup of the sources](https://github.com/fortran-lang/stdlib/blob/master/ci/fpm-deployment.sh) before pushing to the stdlib-fpm branch. I would like to:
> 
> * ask whether there is any interest in it?
> * ask for feedback about whether any of those pre-processing steps could instead be replaced with more advanced source handling in fpm;
> * fpm.toml does not store a `VERSION`, is that by design? otherwise, I'd like to suggest a PR that pushes it on deployment:
> 
> https://github.com/fortran-lang/stdlib/blob/261a6274477ec8e23f94bd8be384158149ed6628/fpm.toml#L1-L10
> 
> ### Prior Art
> _No response_
> 
> ### Additional Information
> _No response_

Yes, long term goal is to build `stdlib` using ``fpm`` without any external build script. So, basically we need to natively support ``fypp`` preprocessing inside fortran package manager.

**Comment by perazz** [comment]
I've added the `stdlib` deployment as a metapackage in `fpm`. It's not merged yet but it can be played around with from [my fork](https://github.com/perazz/fpm/tree/metapackages). Basically, to use stdlib it's enough to turn that on in the fpm manifest as a metapackage: 

```toml
[metapackages]
stdlib = true
```

**Comment by bellomia** [comment]
Since the metapackages feature has been merged in `fpm` (last week, details [here](https://github.com/fortran-lang/fpm/releases/tag/v0.9.0)) the old way of including it as a dependency in the toml manifest does not work anymore.

I.e. a code line as described in the current README would result in a fpm error, for fpm ≥ 0.9.0 https://github.com/fortran-lang/stdlib/blob/0efc112f80e5a4886a8eb5b84a9ffc5146fa6454/README.md?plain=1#L198-L199

Apparently the new  definitive (=merged) way would be

```toml
[dependencies]
stdlib = "*"
```

and I checked it works on my machine, with my fpm package. Though I cannot (have no time to) understand the details of the new way, as much as to propose my own edit to the README. In particular I could not figure out the exact meaning of the asterisk or if it can be replaced with an exact version, from the fpm documentation. @perazz you're probably the one that most efficiently can answer all of this and maybe update the stdlib readme accordingly, so to allow people on the latest fpm release to build packages depending on stdlib without accidents. 

(Feel free to move to a new issue, if the case). :)
